lineman/app/services/form_service.coffee

```diff
@@ -0,0 +1,27 @@
+angular.module('loomioApp').factory 'FormService', ($rootScope, FlashService) ->
+  new class FormService
+
+    errorTypes =
+      400: 'badRequest'
+      401: 'unauthorizedRequest'
+      403: 'forbiddenRequest'
+      404: 'resourceNotFound'
+      422: 'unprocessableEntity'
+      500: 'internalServerError'
+
+    submit: (scope, model, options = {}) ->
+      submitFn = options.submitFn or model.save
+      ->
+        scope.isDisabled = true
+        submitFn(model).then ->
```

Comment:
This is great @gdpelican. I have a few suggestions:

I'm still wondering if all the RecordsInterace methods like deactivate(user) that take the model should just be model instance methods that operate on self.

It would also mean that setErrors can always been handled within the model itself (and only once - chained on .then ) and FormService would not need to be doing that work. I really don't think FormService should be responsible for setting up the model errors.

Also I wonder if the scope.$close for modals should be there.. maybe if ModalService should be responsible for setting its close callback, then FormService does not need to know about modal behaviour.

So if the model took responsiblity for setting it's own errors and modals setup a $close() callback themselves.. then i think this would look really neat. How you do feel about those changes?